server/service/src/invoice_line/stock_out_line/set_prescribed_quantity/mod.rs

```diff
@@ -0,0 +1,338 @@
+use repository::{
+    EqualFilter, InvoiceLine, InvoiceLineFilter, InvoiceLineRepository, InvoiceLineRowRepository,
+    InvoiceLineType, RepositoryError,
+};
+use util::uuid::uuid;
+
+use crate::{invoice_line::query::get_invoice_line, service_provider::ServiceContext};
+
+mod generate;
+use generate::generate;
+mod validate;
+use validate::validate;
+
+#[derive(Clone, Debug, PartialEq, Default)]
+pub struct SetPrescribedQuantity {
+    pub invoice_id: String,
+    pub item_id: String,
+    pub prescribed_quantity: f64,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum SetPrescribedQuantityError {
+    ItemNotFound, // check if valid item id
+    InvoiceDoesNotExist,
+    NotAStockItem,
+    NotAPrescription,
+    NotThisStoreInvoice,
+    NewlyCreatedLineDoesNotExist,
+    DatabaseError(RepositoryError),
+}
+
+impl From<RepositoryError> for SetPrescribedQuantityError {
+    fn from(error: RepositoryError) -> Self {
+        SetPrescribedQuantityError::DatabaseError(error)
+    }
+}
+
+pub fn set_prescribed_quantity(
+    ctx: &ServiceContext,
+    input: SetPrescribedQuantity,
+) -> Result<InvoiceLine, SetPrescribedQuantityError> {
+    let invoice_line = ctx
+        .connection
+        .transaction_sync(|connection| {
+            let item_row = validate(connection, &ctx.store_id, &input)?;
+
+            let invoice_line_row_repo = InvoiceLineRowRepository::new(connection);
+
+            let existing_lines = InvoiceLineRepository::new(connection).query_by_filter(
+                InvoiceLineFilter::new()
+                    .item_id(EqualFilter::equal_to(&input.item_id))
+                    .invoice_id(EqualFilter::equal_to(&input.invoice_id)),
+            )?;
+
+            if existing_lines.len() > 1 {
```

Comment:
I think this logic should have some comments to explain the intended behaviour.

E.g.

When we set prescribed quantity for an item, there's 3 scenarios.

1. We don't have any stock lines already create for that item, in which case we need to create an unallocated line for the prescription to record the prescribed quantity against
2. We already have more than one stock line, we need to update the one that already has prescribed quantity, or assign it to a single line if none of them have it.
3. If there's a proper allocated line, and an unallocated one, we need to remove the unallocated one just to clean up